watson/watson.py

```diff
@@ -134,6 +137,30 @@ def config(self, value):
         self._config = value
         self._config_changed = True
 
+    def _check_default_config(self):
+        if Path(self.config_file).exists():
+            return
+        Path(self.config_file).write_text(textwrap.dedent("""
+            [backend]
+            # url =
+            # token = yourapitoken
+
+            [options]
+            confirm_new_project = false
+            confirm_new_tag = false
+            date_format = %Y.%m.%d
+            log_current = false
+            pager = true
+            report_current = false
+            stop_on_start = false
+            stop_on_restart = false
+            time_format = %H:%M:%S%z
+            week_start = monday
+
+            [default_tags]
+            # some_project = some_tag1 some_tag2
+        """).strip())
```

Comment:
If we had a default config, we actually don't need default value in CLI.

Reply:
I have considered an approach of adding another file `default.cfg` containing this multi-line string, instead of hard-coding it under `watson/watson.py`. But in this way we need to go through [the process](https://stackoverflow.com/questions/1612733/including-non-python-files-with-setup-py) to include non-python files with setup.py. I'm not sure if it's what you want.

Reply:
No, I like the way you generate it, but I think it would be more pythonic (at least more explicit) to declare defaults as code and then generate a default configuration from those.

Reply:
Are you suggesting something like this?
```python
# defaults.py
configs = {
    'options': {
        'confirm_new_project': False,
        # ...
    }
}
```

Reply:
More something like:

```python
# defaults.py
CONFIRM_NEW_PROJECT = False
```

Reply:
I agree that being explicit is always favorable, though I'm not really into this style. One drawback includes that this does not differentiate configs from different sections. And I actually think that a default config itself (e.g. `defaults.cfg` as follows) is more explicit than a config file generated from a module with default variable declarations. As a friendly reminder, we don't need default values in CLI anymore as we already cover them in default config.
```
# defaults.cfg
[options]
confirm_new_project = false
```

Reply:
I get you point, but still, I think using a configuration-file-driven-approach because of the CLI is not ideal. When people are using parts of Watson as a library I think it's better to avoid loading defaults from a configuration file.

Reply:
I get your point if there's future plan to support calling watson api as a library. As currently `watson.watson` has strong coupling with `watson.config`, it's not yet ready for such usage case. Though it's worth some discussion if it's on the blueprint.

IMHO, a better design will be creating a class `WastonOption` serving as a value object (with default values defined), and it can optionally sync with the config file. For instance, in [Watson.add](https://github.com/TailorDev/Watson/blob/f85417177cbec064c93487d7ad8becbc27eb256e/watson/watson.py#L248) it will become
`default_tags = self.options.default_tags` instead of current `default_tags = self.config.getlist('default_tags', project)`. Though I feel that it's out of scope of this PR.

What do you think?

Reply:
Sounds good to me, yay!

Reply:
As this is out of scope of this PR, let's track this in #327. Perhaps we can set aside this issue for now, and reconsider it if #327 or similar designs are implemented someday.